codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsAuthPlugin.java

```diff
@@ -119,6 +122,10 @@ private static boolean testServiceId(Shape serviceShape, String expectedId) {
         return serviceShape.getTrait(ServiceTrait.class).map(ServiceTrait::getSdkId).orElse("").equals(expectedId);
     }
 
+    private static boolean isAwsService(Shape serviceShape) {
+        return serviceShape.getTrait(ServiceTrait.class).isPresent();
+    }
```

Comment:
Can't you just check for sigv4 instead? The `ServiceIndex` had a `getAuthSchemes` operation you can use to generically check for auth. If the problem ist aht you need the aws service trait to make these components work, then you should throw a codegen exception if the sigv4 trait is present but the aws service trait is not.

Reply:
I agree, I think a sigv4 check is more useful here.

Reply:
See the latest diffs. I think the serviceShape.hasTrait(SigV4.class) check is sufficient, so I didn't use ServiceIndex. LMK if you think it is needed.